packages/catalog-client/src/CatalogClient.ts

```diff
@@ -53,12 +56,16 @@ import { DefaultApiClient, TypedResponse } from './generated';
  */
 export class CatalogClient implements CatalogApi {
   private readonly apiClient: DefaultApiClient;
+  private readonly hasQueryEntities: (
+    options?: CatalogRequestOptions,
+  ) => Promise<boolean | undefined>;
 
   constructor(options: {
     discoveryApi: { getBaseUrl(pluginId: string): Promise<string> };
     fetchApi?: { fetch: typeof fetch };
   }) {
     this.apiClient = new DefaultApiClient(options);
+    this.hasQueryEntities = featureDetector(hasQueryEntities(this.apiClient));
```

Comment:
If we actually want this I'd say we should implement it through an introspection endpoint rather than a detector. I'd note that down as a potential backend system feature and ship without the feature detection logic now though. The query entities endpoint has been out for a long while, and we ofc explicitly require backends to be bumped first.

We don't have a strict policy around versioning of separately deployed backends though, bit of a gap. Gonna make a note of that, maybe an issue 😁